articles/openshift/oshift-how-configure-portworx-volume-replication.md

```diff
@@ -0,0 +1,139 @@
+services: openshift
+author: Ben Bacon
+reviewer:
+lastreviewed: 14/10/2019
+toc_rootlink: How To
+toc_sub1: Use Portworx with OpenShift
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Configure Portworx volume replication
+toc_fullpath: How To/Use Portworx with OpenShift/oshift-how-configure-portworx-volume-replication.md
+toc_mdlink: oshift-how-configure-portworx-volume-replication.md
+---
+
+# How to configure Portworx volume replication
+
+## Overview
+
+Portworx is a cloud-native storage solution that is now available as an integration with our OpenShift offering. This article describes how to configure replication for Portworx volumes and provides an overview of Portworx failover capabilities.
+
+### Intended audience
+
+This article assumes you have access to a Portworx integrated OpenShift 3.11 cluster and that you have cluster-admin rights. It also assumes familiarity with `oc`, the OpenShift command-line client. 
+
+If you're interested in a free 30 day trial of Portworx, raise a Service Request via the [My Calls](https://portal.skyscapecloud.com/support/ivanti) section of the UKCloud Portal.
+
+## Replication
+
+Portworx runs as a container on all compute nodes in a cluster and stores volume data on block storage attached to each of these nodes. This volume data can be synchronously replicated between nodes to provide redundancy and increase read speed using parallelism across nodes.
+
+### Configuring the replication factor
+
+The replication factor of a volume is determined by the storage class from which a persistent volume claim is created. Use the following parameter to set the replication factor to a value between 1 and 3 within the storage class spec:
+
+```none
+parameters:
+   repl: "2"
+```
+
+If a pod is running on a node that becomes isolated, the pod can be scheduled to an alternate node provided there is a replica on that same node.
+
+The following example shows the spec for a storage class with a replication factor of 3:
+
+```none
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+    name: portworx-repl3
+provisioner: kubernetes.io/portworx-volume
+parameters:
+   repl: "3"
+```
+
+### Modifying a volume replication factor
+
+You can increase or decrease the replication factor for a Portworx volume after it has been created using the `pxctl` command.
+
+1. Before proceeding, identify the volume name for the persistent volume claim:
+
+    ```none
+    $ oc get pvc | awk '{print $1" "$3}'
+    NAME VOLUME
+    pvc-repl2 pvc-e15d82b2-ed46-11e9-8422-fa163e52fd0e
+    ```
+
+2. Verify the current replication factor and the nodes storing the replicas, substituting the volume name returned in the previous step:
+
+    ```none
+    $ PX_POD=$(oc get pods -l name=portworx -n kube-system -o jsonpath='{.items[0].metadata.name}')
+    
+    $ oc exec $PX_POD -n kube-system -c portworx -- /opt/pwx/bin/pxctl volume inspect pvc-e15d82b2-ed46-11e9-8422-fa163e52fd0e | grep -E 'HA|Node'
```

Comment:
You go on to use `oc exec` a lot in this article - it's probably worth calling out exactly what you're doing when you run this command and why you're using it to access the pxctl commands. Might be worth a separate section or a tip or similar box.